client/src/main/kotlin/no/nav/su/se/bakover/client/stubs/oppdrag/SimuleringStub.kt

```diff
@@ -71,41 +38,202 @@ class SimuleringStub(val clock: Clock) : SimuleringClient {
                 .sumOf { it.belop }
         }
 
-    private fun simulerIngenUtbetaling(utbetaling: Utbetaling): Simulering {
-        val simuleringsPeriode = when (val sisteUtbetalingslinje = utbetaling.sisteUtbetalingslinje()) {
-            is Utbetalingslinje.Endring -> SimulertPeriode(
-                fraOgMed = sisteUtbetalingslinje.virkningstidspunkt,
-                tilOgMed = utbetaling.senesteDato(),
-                utbetaling = emptyList(),
-            )
-            else -> SimulertPeriode(
-                fraOgMed = utbetaling.tidligsteDato(),
-                tilOgMed = utbetaling.senesteDato(),
-                utbetaling = emptyList(),
+    private fun simulerUtbetalinger(utbetaling: Utbetaling): Simulering {
+        return utbetaling.utbetalingslinjer.map {
+            when (it) {
+                is Utbetalingslinje.Endring -> {
+                    Periode.create(it.virkningstidspunkt, it.tilOgMed)
```

Comment:
Burde det være minOf(it.virkningstidspunkt,it.fraOgMed)?

Reply:
Det blir riktig å bruke virkningstidspunkt siden det er denne datoen som er avgjørende for tidspunktet endringen får effekt. FraOgMed er bare linjen som endres sin opprinnelige dato, virkningstidspunktet for endringen kan være både før, på og etter denne datoen.